rtil/src/threads/ue/rebo/rebo_init.rs

```diff
@@ -404,8 +423,22 @@ fn list_recordings() -> Vec<String> {
         }).collect()
 }
 #[rebo::function("Tas::save_recording")]
-fn save_recording(filename: String, recording: Vec<RecordFrame>) {
-    let filename = sanitize_filename::sanitize(filename);
+fn save_recording(filename: String, frames: Vec<RecordFrame>, recording_start_timestamp: u64, recording_end_timestamp: u64) {
+    let recording = Recording {
+        version: 1,
+        author: AMyCharacter::get_player().get_player_name(),
+        steam_id: AMyCharacter::get_player().get_steamid(),
+        filename: filename.clone(),
+        frame_count: frames.len() as i64,
+        recording_start_timestamp,
+        recording_end_timestamp,
+        recording_save_timestamp: SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_millis() as u64,
+        base_speed: AMyCharacter::get_base_speed(),
+        max_walk_speed: AMyCharacter::get_max_walk_speed(),
+        max_bonus_speed: AMyCharacter::get_max_bonus_speed(),
+        frames,
+    };
+    let filename = sanitize_filename::sanitize(filename.clone());
```

Comment:
No need to clone, `sanitize` takes `impl AsRef<str>` which `&String` implements.